source.go

```diff
@@ -451,7 +473,13 @@ func PatchSources(worker llb.State, spec *Spec, sourceToState map[string]llb.Sta
 			continue
 		}
 		opts = append(opts, ProgressGroup("Patch spec source:"+sourceName))
-		states[sourceName] = patchSource(worker, sourceState, states, patches, withConstraints(opts))
+
+		var nestedPath string
+		if SourceIsDir(spec.Sources[sourceName]) {
```

Comment:
This is new behavior that is not mimicked in the rpm patcher.
Was there a particular reason to add it?

Reply:
I think the reason it was added here (correct me if I'm wrong) is because `specToSourcesLLB` takes the extra step of nesting a source within a directory in the LLB itself. I.e., the LLB contains the directory as well as the contents. I think that nesting step could be moved out of `spectToSourcesLLB` (if indeed it is needed), so we don't have to handle that behavior here

Reply:
@adamperlin yes that's correct. I went back and changed `specToSourcesLLB` so that it doesn't nest the sources in the directories. Now that's happening in the `RunOpt` that mounts the sources. I restored the two patching functions (the public and private) to the original versions. This is much cleaner. Thanks for pointing this out @cpuguy83.

This is done

Reply:
This is not done. Working on it now.

Reply:
Now this is done.